### DISCUSSION TITLE
Status message sometimes wrong?

### DISCUSSION THREAD

**Discussion by camueller** [meatpiHQ/wican-fw#181] [General]
[I use WiCAN to retrieve the SOC of my EV when it comes near my house. The software waits for the an "online" message from the WiCAN in order to send a CAN message to retrieve the SOC.](https://github.com/camueller/SmartApplianceEnabler/tree/master/run/soc/wican)

Yesterday evening (17:25:57 aka 5:25:57 PM) I left the house and returned shortly after midnight (00:02:51 AM). The WiCAN sent a status message "offline" (which I assume should have been "online"). Afterwards at 00:06:33 AM it sent another status message "offline" - probably to indicate that the cars controllers have been turned off after I switched the car itself off.

```
Aug 23 17:24:57 raspi2 wican-status.sh[27036]: Sleeping ...
Aug 23 17:25:57 raspi2 wican-status.sh[27036]: Waiting for message ...
Aug 24 00:02:51 raspi2 wican-status.sh[27036]: Message received:
Aug 24 00:02:51 raspi2 wican-status.sh[24962]: {"status": "offline"}
Aug 24 00:02:51 raspi2 wican-status.sh[27036]: Sleeping ...
Aug 24 00:03:51 raspi2 wican-status.sh[27036]: Waiting for message ...
Aug 24 00:06:33 raspi2 wican-status.sh[27036]: Message received:
Aug 24 00:06:33 raspi2 wican-status.sh[25024]: {"status": "offline"}
Aug 24 00:06:33 raspi2 wican-status.sh[27036]: Sleeping ...
Aug 24 00:07:33 raspi2 wican-status.sh[27036]: Waiting for message ...
```

Since there was no "online" message in this case the SOC has not been retrieved and the energy management system of my house assumes the SOC to be still at 100% which is wrong.

**Comment by camueller** [comment]
The problem occured again ...

**Reply by ccremer224** [reply]
In my e-Golf, the WiCAN also is online periodically and sends the 12V accu voltage. The SOC only is achievable if the CAN Bus from the car is active. This only is the case if the car itself is switched on and the CAN bus stays active for some time after having turned off the car.

**Reply by camueller** [reply]
The log above does not show a situation caused by the WiCAN starting up by accident or periodically. I was driving the car and then turned it of. Therefore the first "offline" message should have been an "online" message instead.

**Comment by meatpiHQ** [comment]
@camueller The offline message is a mqtt last will message. It is sent when the device disconnects from the mqtt broker could be because of bad wifi signal.

**Reply by camueller** [reply]
Which one? Both? I don't think it is possible to send 2 mqtt last will without having to come online in between. So there must be an error somewhere ...

**Reply by meatpiHQ** [reply]
Yes that's right, it's coming online then offline. Only the offline is the last will message.

Do you have sleep mode enabled?

**Reply by camueller** [reply]
Yes

**Comment by typxxi** [comment]
We have found only 1 solution to stop the alarm of the e-golf turning on and transmitting data at arrival:

1. turn the ignition off and on for 10 seconds - if you only arrive and expect that the canbus data are transferred at least we had no luck with a MY 2020 golf from the last production year
2. to prevent the alarm to turn on somewhere later mostly in the night you have to leave the car, close the door and and then open the door to close it again

This way the car stops autolocking and stays open in  your garage without an alarm. Not great but usefull. 
If you do not open it the car migh turn on the alarm from time to time with an unpredictable delay, sometimes hours later, sometimes even days later. So the car has to be open while connected with the wallbox, at least here.
